src/internal/storage/kv/memcache.go

```diff
@@ -0,0 +1,66 @@
+package kv
+
+import (
+	"context"
+	"sync"
+
+	"github.com/hashicorp/golang-lru/simplelru"
+)
+
+type memoryCache struct {
+	mu    sync.Mutex
+	cache *simplelru.LRU
+}
+
+// NewMemCache returns a new in memory cache
+func NewMemCache(size int) GetPut {
+	mc := &memoryCache{}
+	var err error
+	mc.cache, err = simplelru.NewLRU(size, mc.onEvict)
+	if err != nil {
+		panic(err)
+	}
+	return mc
+}
+
+func (mc *memoryCache) Put(ctx context.Context, key, value []byte) error {
+	k := string(key)
+	v := append([]byte{}, value...)
+	mc.mu.Lock()
+	defer mc.mu.Unlock()
+	mc.cache.Add(k, v)
+	return nil
+}
+
+func (mc *memoryCache) GetF(ctx context.Context, key []byte, cb ValueCallback) error {
+	v := mc.get(key)
+	if v == nil {
+		return ErrKeyNotFound
+	}
+	return AssertNotModified(v, cb)
+}
+
+func (mc *memoryCache) Delete(ctx context.Context, key []byte) error {
+	mc.mu.Lock()
+	defer mc.mu.Unlock()
+	mc.cache.Remove(key)
+	return nil
+}
+
+func (mc *memoryCache) Exists(ctx context.Context, key []byte) (bool, error) {
+	v := mc.get(key)
+	return v == nil, nil
+}
+
+func (mc *memoryCache) get(key []byte) []byte {
+	mc.mu.Lock()
+	defer mc.mu.Unlock()
+	v, ok := mc.cache.Get(string(key))
+	if !ok {
+		return nil
+	}
+	return v.([]byte)
+}
+
+func (mc *memoryCache) onEvict(key, value interface{}) {
```

Comment:
I think we should just pass in nil for the `onEvict` function in `NewLRU` since we are not using it. Then get rid of this.